autorest/azure/azure.go

```diff
@@ -50,8 +51,8 @@ type RequestError struct {
 
 // Error returns a human-friendly error message from service error.
 func (e RequestError) Error() string {
-	return fmt.Sprintf("azure: Service returned an error. Code=%q Message=%q Status=%d",
-		e.ServiceError.Code, e.ServiceError.Message, e.StatusCode)
+	return fmt.Sprintf("autorest/azure: Service returned an error. Code=%q Message=%q Details=%v Status=%v",
+		e.ServiceError.Code, e.ServiceError.Message, e.ServiceError.Details, e.StatusCode)
```

Comment:
ServiceError is response error object. ``` [error: code, message, details] ```
RequestError is ```ServiceError + RequestID + DetailedError(package name, statuscode, Method, message)```. I think we can remove ServiceError.Error() because we never refer to ServiceError directly. It always comes wrapped inside RequestError.